src/applications/facility-locator/components/ResultsList.jsx

```diff
@@ -37,6 +41,67 @@ class ResultsList extends Component {
     }
   }
 
+  /**
+   * Returns Result items by type
+   * @param isMobile boolean
+   * @param query object
+   * @param results array list
+   * @returns [] list of results
+   */
+  // this approach is to refactor result items to components of its own kind and use common components when needed
+  // reason is that we have been getting requirements to modify result elements on a regular basis (add notices, delete sentences, update copy)
+  // and the checks are getting too repetitive to do it in each components of the SearchResult.jsx file
+  // eventually we can remove SearchResult.jsx completely and handle render logic here
+  renderResultItems(isMobile, query, results) {
+    return results.map(r => {
+      let item;
+      switch (query.facilityType) {
+        case 'health':
+        case 'cemetery':
+        case 'benefits':
+        case 'vet_center':
+          item = <VaFacilityResult location={r} query={query} />;
+          break;
+        case 'provider':
+        case 'cc_provider':
+          // Support non va urgent care search through ccp option
+          if (query.serviceType === CLINIC_URGENTCARE_SERVICE) {
+            item = <UrgentCareResult provider={r} query={query} />;
+          } else {
+            item = <CCProviderResult provider={r} query={query} />;
+          }
+          break;
+        case 'pharmacy':
+        case 'cc_pharmacy':
+          item = <PharmacyResult provider={r} query={query} />;
+          break;
+        case 'urgent_care':
+          if (query.serviceType === 'NonVAUrgentCare') {
+            item = <UrgentCareResult provider={r} query={query} />;
+          } else if (!query.serviceType || query.serviceType === 'UrgentCare') {
+            // this is probably going to change, it should default to All,
+            // currently the logic is if serviceType null, the result is VA urgent care
```

Comment:
This works, but it's kind of fragile. I would prefer to set the default serviceType in the searchQuery reducer. The UI should match the query.